astropy/units/utils.py

```diff
@@ -68,6 +68,9 @@ def _iter_unit_summary(
         if not isinstance(val, core.UnitBase):
             continue
 
+        if not isinstance(val, core.NamedUnit):
```

Comment:
Should the typing be adjusted, in that we now are guaranteed to return a `Generator[tuple[NamedUnit, ...`?

Reply:
Of course, we could also just *skip* non-`NamedUnit` -- but I think what you have makes a lot more sense.

Reply:
The return annotation should indeed be updated.